### PR TITLE
[kvutils] - Add errors for post execution conflicts and out of time bounds [KVL-1369]

### PR DESCRIPTION
**Pull request by nicu-da**
changelog_begin
changelog_end

### CODE REVIEW

File: ledger/error/src/main/scala/com/daml/error/ErrorCategory.scala [hunk 1]
```diff
@@ -360,6 +361,23 @@ object ErrorCategory {
...
+  object InconsistentSystemStateDuringSubmission
+      extends ErrorCategoryImpl(
+        grpcCode = Some(Code.ABORTED),
+        logLevel = Level.INFO,
+        retryable = None,
+        securitySensitive = false,
+        asInt = 15,
+        rank = 3,
+      )
+      with ErrorCategory
```
Comment (fabiotudone-da) [inline ledger/error/src/main/scala/com/daml/error/ErrorCategory.scala:379]: The only concern I have with this is that so far we have one category per gRPC status code we decided to use. I think we should reuse `ContentionOnSharedResources`.

Comment (nicu-da) [reply ledger/error/src/main/scala/com/daml/error/ErrorCategory.scala]: Done

File: ledger/participant-state-kv-errors/src/main/scala/com/daml/ledger/error/definitions/kv/KvErrors.scala [hunk 2]
```diff
@@ -90,6 +110,24 @@ object KvErrors extends ErrorGroup()(ErrorGroups.rootErrorClass) {
       }
     }
 
+    @Explanation(
+      "The record time is not within bounds for reasons such as " +
+        "excessive latency. Excessive clock skew between the participant and the committer " +
+        "or a time model that is too restrictive may also produce this rejection."
```
Comment (tudor-da) [inline ledger/participant-state-kv-errors/src/main/scala/com/daml/ledger/error/definitions/kv/KvErrors.scala:116]: ?
```suggestion
        "excessive latency, excessive clock skew between the participant and the committer " +
        "or a time model that is too restrictive."
```

**Review by tudor-da — APPROVED** [review]

**Review by fabiotudone-da — APPROVED** [review]

File: ledger/participant-state-kv-errors/src/main/scala/com/daml/ledger/error/definitions/kv/KvErrors.scala [hunk 3]
```diff
@@ -90,6 +110,24 @@ object KvErrors extends ErrorGroup()(ErrorGroups.rootErrorClass) {
...
+        "or a time model that is too restrictive."
+    )
+    @Resolution("Retry the submission or contact the participant operator.")
+    object RecordTimeOutOfBounds
+        extends ErrorCode(
+          id = "RECORD_TIME_OUT_OF_BOUNDS",
+          ErrorCategory.ContentionOnSharedResources, // It may succeed at a later time
+        ) {
+      case class Reject(
+          override val definiteAnswer: Boolean,
```
Comment (miklos-da) [inline ledger/participant-state-kv-errors/src/main/scala/com/daml/ledger/error/definitions/kv/KvErrors.scala:125]: Why do we need to override this field?

Comment (nicu-da) [reply ledger/participant-state-kv-errors/src/main/scala/com/daml/ledger/error/definitions/kv/KvErrors.scala:125]: Good point, I had the wrong impression that it's an abstract. Will open a short PR to remove it

Comment (nicu-da) [reply ledger/participant-state-kv-errors/src/main/scala/com/daml/ledger/error/definitions/kv/KvErrors.scala:125]: https://github.com/digital-asset/daml/pull/13728

**Pull request merged**